etcdserver/v3_server.go

```diff
@@ -802,3 +802,14 @@ func (s *EtcdServer) linearizableReadNotify(ctx context.Context) error {
 		return ErrStopped
 	}
 }
+
+func (s *EtcdServer) AuthInfoFromCtx(ctx context.Context) (*auth.AuthInfo, error) {
+	if s.Cfg.ClientCertAuthEnabled {
```

Comment:
@xiang90 is there any reason to keep this as optional in v3 if it'll just fall back to regular user auth?